src/security/kuzzleUser.js

```diff
@@ -0,0 +1,134 @@
+var
+  KuzzleSecurityDocument = require('./kuzzleSecurityDocument'),
+  KuzzleProfile = require('./kuzzleProfile');
+
+function KuzzleUser(kuzzleSecurity, id, content) {
+
+  KuzzleSecurityDocument.call(this, kuzzleSecurity, id, content);
+
+  // Hydrate user with profile if profile is not only a string but an object with `_id` and `_source`
+  if (content.profile && content.profile._id && content.profile._source) {
+    content.profile = new KuzzleProfile(kuzzleSecurity, content.profile._id, content.profile._source);
+  }
+
+  // Define properties
+  Object.defineProperties(this, {
+    // private properties
+    deleteActionName: {
+      value: 'deleteUser'
+    }
+  });
+
+  // promisifying
+  if (kuzzleSecurity.kuzzle.bluebird) {
+    return kuzzleSecurity.kuzzle.bluebird.promisifyAll(this, {
+      suffix: 'Promise',
+      filter: function (name, func, target, passes) {
+        var whitelist = ['hydrate', 'save'];
+
+        return passes && whitelist.indexOf(name) !== -1;
+      }
+    });
+  }
+}
+
+KuzzleUser.prototype = Object.create(KuzzleSecurityDocument.prototype, {
+  constructor: {
+    value: KuzzleUser
+  }
+});
+
+/**
+ * This function allow to get the hydrated user of the corresponding current user.
+ * The hydrated user has profiles and roles.
+ *
+ * @param {responseCallback} [cb] - Handles the query response
+ */
+KuzzleUser.prototype.hydrate = function (cb) {
+  var
+    self = this;
+
+  self.kuzzle.callbackRequired('KuzzleUser.hydrate', cb);
+
+  if (!this.content.profile || typeof this.content.profile !== 'string') {
+    throw new Error('The User must contains a profile as string in order to be hydrated');
+  }
+
+  self.kuzzle.query(this.kuzzleSecurity.buildQueryArgs('getProfile'), {_id: this.content.profile}, null, function (error, response) {
+    if (error) {
+      return cb(error);
+    }
+
+    cb(null, new KuzzleUser(self, response.result._id, response.result._source));
+  });
+};
+
+/**
+ * Set profile in content
+ * @param {KuzzleProfile|string} profile - can be a KuzzleProfile or an id string
+ *
+ * @returns {KuzzleUser} this
+ */
+KuzzleUser.prototype.setProfile = function (profile) {
+
+  if (typeof profile !== 'string' && !(profile instanceof KuzzleProfile)) {
+    throw new Error('Parameter "profile" must be a KuzzleProfile or a string');
+  }
+
+  this.content.profile = profile;
+
+  return this;
+};
+
+/**
+ * Saves this user into Kuzzle.
+ *
+ * If this is a new user, this function will create it in Kuzzle.
+ * Otherwise, this method will replace the latest version of this user in Kuzzle by the current content
+ * of this object.
+ *
+ * @param {responseCallback} [cb] - Handles the query response
+ *
+ * @returns {*} this
+ */
+KuzzleUser.prototype.save = function (cb) {
+  var
+    data = this.serialize(),
+    self = this;
+
+  self.kuzzle.query(this.kuzzleSecurity.buildQueryArgs('createOrReplaceUser'), data, null, function (error) {
```

Comment:
To avoid any warning raised by your IDE, skipping the `options` argument altogether should do the trick: I made this argument optional in the JSDoc of the `query` method.

But another approach would be to add an optional `options` argument to this method (and maybe other methods too). Because by default, nearly all methods should support the `queuable` option, asking to (not) queue this query during offline mode.
The only exception to this rule I can think of should be the `login` and `logout` functions (and all other functions not sending any requests, like `addListener`, obviously).